### PR TITLE
feat(provider/kubernetes): v2 Add support for api service kind

### PR DESCRIPTION
**Pull request by Tizaniy**
issue: https://github.com/spinnaker/spinnaker/issues/2894

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesAPIServiceHandler.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import com.netflix.spinnaker.clouddriver.model.Manifest.Status;
+import org.springframework.stereotype.Component;
+
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.op.handler.KubernetesHandler.DeployPriority.ROLE_PRIORITY;
+
+@Component
+public class KubernetesAPIServiceHandler extends KubernetesHandler {
+  @Override
+  public int deployPriority() {
+    return ROLE_PRIORITY.getValue();
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesAPIServiceHandler.java]: Is this correct? It means it's deployed at the same priority as roles & cluster roles for RBAC

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesAPIServiceHandler.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  @Override
+  public KubernetesKind kind() {
+    return KubernetesKind.API_SERVICE;
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesAPIServiceHandler.java]: What version of k8s is needed for this resource?

**Comment by lwander** [comment]
Thanks @Tizaniy!

**Comment by Tizaniy** [comment]
APIService deploy priority was changed

**Comment by Tizaniy** [comment]
Please review: @spinnaker/reviewers

**Pull request merged**